testsuite/MDAnalysisTests/coordinates/test_chemfiles.py

```diff
@@ -0,0 +1,152 @@
+        self.writer = ChemfilesWriter
+        self.ext = 'xyz'
+        self.volume = 0
+        self.dimensions = np.zeros(6)
+        self.dimensions[3:] = 90.0
+
+
+class TestChemfilesReader(MultiframeReaderTest):
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return ChemfilesXYZReference()
+
+
+class TestChemfilesWriter(BaseWriterTest):
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return ChemfilesXYZReference()
+
+    # Disable 'test_no_container' as it try to open a file for writing without
+    # extension.
+    def test_no_container(self, ref):
+        pass
+
+
+class TestChemfiles(TestCase):
+    def test_read_chemfiles_format(self):
+        u = mda.Universe(
+            datafiles.LAMMPSdata,
+            format="chemfiles",
+            topology_format="data",
+            chemfiles_format="LAMMPS Data"
+        )
+
+        for ts in u.trajectory:
+            self.assertEqual(ts.n_atoms, 18364)
+
+    def test_changing_system_size(self):
+        outfile = tempdir.TempDir().name + "chemfiles-changing-size.xyz"
+        with open(outfile, "w") as fd:
+            fd.write(VARYING_XYZ)
+
+        u = mda.Universe(outfile, format="chemfiles", topology_format="XYZ")
+
+        with self.assertRaises(IOError):
+            u.trajectory._read_next_timestep()
+
+    def test_wrong_open_mode(self):
+        with self.assertRaises(IOError):
+            _ = ChemfilesWriter("", mode="r")
+
+    def check_topology(self, reference, file):
+        u = mda.Universe(reference)
+        atoms = set([
+            (atom.name, atom.type, atom.record_type)
+            for atom in u.atoms
+        ])
+        bonds = set([
+            (bond.atoms[0].ix, bond.atoms[1].ix)
+            for bond in u.bonds
+        ])
+
+        check = mda.Universe(file)
+        np.testing.assert_equal(
+            u.trajectory.ts.positions,
+            check.trajectory.ts.positions
+        )
+
+        for atom in check.atoms:
+            self.assertIn((atom.name, atom.type, atom.record_type), atoms)
+
+        for bond in check.bonds:
+            self.assertIn((bond.atoms[0].ix, bond.atoms[1].ix), bonds)
+
+    def test_write_topology(self):
+        u = mda.Universe(datafiles.CONECT)
+        outfile = tempdir.TempDir().name + "chemfiles-write-topology.pdb"
+        with ChemfilesWriter(outfile) as writer:
+            writer.write(u)
+        self.check_topology(datafiles.CONECT, outfile)
+
+        # Manually setting the topology when creating the ChemfilesWriter
+        # (1) from an object
+        with ChemfilesWriter(outfile, topology=u) as writer:
+            writer.write_next_timestep(u.trajectory.ts)
+        self.check_topology(datafiles.CONECT, outfile)
+
+        # (2) from a file
+        with ChemfilesWriter(outfile, topology=datafiles.CONECT) as writer:
+            writer.write_next_timestep(u.trajectory.ts)
+        # FIXME: this does not work, since chemfiles also insert the bonds
```

Comment:
Is this is a bug in MDAnalysis?

Reply:
I'll have to check again what is going on here, this is not very fresh in my head =)

Reply:
Ok, so the idea is that the PDB format contains "implicit" bonds between the standard residues (the peptide bonds I think), which are automatically added by chemfiles (see https://github.com/chemfiles/chemfiles/blob/da4c316f6e494aa5d0a78b2554eee641b8b79119/src/formats/PDB.cpp#L473), while MDA only reads uses the CONNECT records from the file. 

If you look at the CONNECT specification (http://www.wwpdb.org/documentation/file-format-content/format33/sect10.html#CONECT), you have the following (emphasis is mine):

> The CONECT records specify connectivity between atoms for which coordinates are supplied. The connectivity is described using the atom serial number as shown in the entry. CONECT records are mandatory for HET groups (excluding water) and **for other bonds not specified in the standard residue connectivity table**. These records are generated automatically. 

I don't know if you consider this to be a bug from the MDA side, I guess it depends on how much you want to follow the standard, especially considering that the standard residue connectivity table is huge (50MB http://ftp.wwpdb.org/pub/pdb/data/monomers/het_dictionary.txt). On the Chemfiles side, we don't add bonds for all the monomers, only amino acids since doing otherwise would crash the compiler ¯\_(ツ)_/¯

Reply:
Would be an interesting bug report. The other @MDAnalysis/coredevs are in a better position to decide if this is a *bug*.

Reply:
#2526